### PR TITLE
feat(telemetry): adds reads and writes

### PR DESCRIPTION
**Pull request by praveen-influx**
- instrumented code to get read and write measurement
- introduced EventsBucket for collection of reads/writes
- sampler now samples every minute for all metrics (including reads/writes)
- other tidy ups

closes: https://github.com/influxdata/influxdb/issues/25372

### CODE REVIEW

File: influxdb3_server/src/http.rs [hunk 1]
```diff
@@ -419,6 +419,13 @@ where
                 .await?
         };
 
+        let num_lines = result.line_count;
+        let payload_size = body.len();
+        let telem_store = Arc::clone(&self.common_state.telemetry_store);
+        tokio::spawn(async move {
```
Comment (pauldix) [inline influxdb3_server/src/http.rs:425]: I don't think we want to spawn on every write call. We should be able to just call add_write_metrics on the store without this.

File: influxdb3_server/src/query_executor.rs [hunk 2]
```diff
@@ -146,6 +151,13 @@ impl QueryExecutor for QueryExecutorImpl {
         let token = token.permit();
 
         debug!("execute stream of query results");
+
+        let telem_clone = Arc::clone(&self.telemetry_store);
+        tokio::spawn(async move {
```
Comment (pauldix) [inline influxdb3_server/src/query_executor.rs:156]: Same as before, we don't want to spawn on every query.

File: influxdb3_telemetry/Cargo.toml [hunk 3]
```diff
@@ -17,6 +17,8 @@ parking_lot.workspace = true
 sysinfo.workspace = true
 num.workspace = true
 thiserror.workspace = true
+tonic.workspace = true
+tower.workspace = true
```
Comment (hiltontj) [inline influxdb3_telemetry/Cargo.toml:21]: Are these used since moving away from the middleware/interceptor approach?

**Review by hiltontj — APPROVED** [review]
Just one small thing for deps cleanup.

**Review by pauldix — CHANGES_REQUESTED** [review]

**Review by pauldix — APPROVED** [review]

**Pull request merged**